File: java-sdk/com/semantria/objects/output/DocAnalyticData.java
package com.semantria.objects.output;

import java.util.List;
import javax.xml.bind.annotation.*;
@XmlRootElement(name="document")
public final class DocAnalyticData 
{
	private String id;
	private String config_id;
	private TaskStatus status;
	private Float sentiment_score;
	private List<DocEntity> entities;
	private List<DocTopic> topics;
	private List<DocTheme> themes;
    private List<DocPhrase> phrases;
	private String summary;
	
	@XmlElement(name="id")
	public String getId() { return id; }
	@XmlElement(name="config_id")
	public String getConfigId() { return config_id; }
	@XmlElement(name="status")
	public TaskStatus getStatus() { return status; }
	@XmlElement(name="sentiment_score")
	public Float getSentimentScore() { return sentiment_score; }
	@XmlElementWrapper(name="entities")
	@XmlElement(name="entity")
	public List<DocEntity> getEntities() { return entities; }
	@XmlElementWrapper(name="topics")
	@XmlElement(name="topic")
	public List<DocTopic> getTopics() { return topics; }
	@XmlElementWrapper(name="themes")
	@XmlElement(name="theme")
	public List<DocTheme> getThemes() { return themes; }
    @XmlElementWrapper(name="phrases")
    @XmlElement(name="phrase")
    public List<DocPhrase> getPhrases() { return phrases; }
	@XmlElement(name="summary")
	public String getSummary() { return summary; }
	
	public void setId(String aid) { id = aid;}
	public void setConfigId(String confId) { config_id = confId; }
	public void setStatus(TaskStatus astatus) { status = astatus; }
	public void setSentimentScore(Float ascore) { sentiment_score = ascore; }
	public void setEntities(List<DocEntity> sentities) { entities = sentities; }
	public void setTopics(List<DocTopic> atopics) { topics = atopics; }
	public void setThemes(List<DocTheme> athemes) { themes = athemes; }
    public void setPhrases(List<DocPhrase> phrases) { this.phrases = phrases; }
	public void setSummary(String asummary) { summary = asummary; }
	
}

File: cpp-sdk/src/objects/output/DocAnalyticData.cpp
#include "DocAnalyticData.h"
#include "../../serializers/xml/DocumentXmlHandler.h"

DocAnalyticData::DocAnalyticData() {
    entities = new vector<DocEntity*>();
    topics = new vector<DocTopic*>();
    themes = new vector<DocTheme*>();
    phrases = new vector<DocPhrase*>();
}

DocAnalyticData::~DocAnalyticData() {
    delete entities;
    delete topics;
    delete themes;
    delete phrases;
}

string DocAnalyticData::GetStatusAsString() {
    return EnumsHelper::GetStatusAsString(status);
}

void DocAnalyticData::SetStatusFromString(string status) {
    this->status = EnumsHelper::GetStatusFromString(status);
}

void DocAnalyticData::Serialize(Json::Value& root) {}

void DocAnalyticData::Deserialize(Json::Value& root) {
    id = root.get("id", "").asString();
    config_id = root.get("config_id", "").asString();
    summary = root.get("summary", "").asString();
    sentiment_score = root.get("sentiment_score", 0.0).asDouble();
    this->SetStatusFromString(root.get("status", "").asString());

    // Entities
    if (NULL == this->entities) {
        this->entities = new vector<DocEntity*>();
    }

    Json::Value entities = root["entities"];
    for ( int i = 0; i < entities.size(); ++i ) {
        DocEntity* entity = new DocEntity();
        entity->Deserialize(entities[i]);
        this->entities->push_back(entity);
    }

    // Topics
    if (NULL == this->topics) {
        this->topics = new vector<DocTopic*>();
    }

    Json::Value topics = root["topics"];
    for ( int i = 0; i < topics.size(); ++i ) {
        DocTopic* topic = new DocTopic();
        topic->Deserialize(topics[i]);
        this->topics->push_back(topic);
    }

    // Themes
    if (NULL == this->themes) {
        this->themes = new vector<DocTheme*>();
    }

    Json::Value themes = root["themes"];
    for ( int i = 0; i < themes.size(); ++i ) {
        DocTheme* theme = new DocTheme();
        theme->Deserialize(themes[i]);
        this->themes->push_back(theme);
    }

    // Phrases
    if (NULL == this->phrases) {
        this->phrases = new vector<DocPhrase*>();
    }

    Json::Value phrases = root["phrases"];
    for ( int i = 0; i < phrases.size(); ++i ) {
        DocPhrase* phrase = new DocPhrase();
        phrase->Deserialize(phrases[i]);
        this->phrases->push_back(phrase);
    }
}

void DocAnalyticData::Deserialize(std::string source) {
    xmlSAXUserParseMemory(GetXmlHandler(), this, source.c_str(), int(source.length()));
    xmlCleanupParser();
}

void DocAnalyticData::Serialize(xmlNodePtr root) {}

xmlSAXHandler* DocAnalyticData::GetXmlHandler() {
    xmlSAXHandler* result = new xmlSAXHandler();
    result->startElement = &DocumentXmlHandler::startElement;
    result->endElement = &DocumentXmlHandler::endElement;
    result->characters = &DocumentXmlHandler::characters;

    return result;
}

string DocAnalyticData::GetRootElement() {
    return "document";
}
